source/js/foundation/pages/youtube-regrets/regrets-reporter-timeline.js

```diff
@@ -0,0 +1,91 @@
+import utility from "../../../utility.js";
+
+let elements = {
+  timeline: `#view-youtube-regrets-reporter .timeline-wrapper .timeline`,
+  segments: `#view-youtube-regrets-reporter .timeline-wrapper .timeline .segment`,
+  button: `#view-youtube-regrets-reporter .timeline-wrapper .btn-timeline-next`,
+};
+
+class RegretsReporterTimeline {
+  constructor() {
+    this.buttonClickCounter = 0;
+
+    this.init();
+  }
+
+  /**
+   * Check if segment is fully shown in current viewport
+   */
+  checkIfInView(seg) {
+    let windowWidth = window.innerWidth || document.documentElement.clientWidth;
+
+    return seg.getBoundingClientRect().right <= windowWidth;
+  }
+
+  /**
+   * Update each timeline segment's opacity
+   */
+  updateOpacity() {
+    elements.segments.forEach((seg) => {
+      if (this.checkIfInView(seg)) {
+        seg.classList.remove(`faded`);
+      } else {
+        seg.classList.add(`faded`);
+      }
+    });
+  }
+
+  /**
+   * Initiate RegretsReporter timeline
+   */
+  init() {
+    this.loadAttempt = this.loadAttempt || 1;
+
+    if (!utility.checkAndBindDomNodes(elements, true)) {
+      if (this.loadAttempt++ < 5) {
+        setTimeout(() => this.init(), 200);
+      }
+      return;
+    }
+
+    this.timeline = elements.timeline[0];
+
+    this.updateOpacity();
+
+    this.timeline.addEventListener("transitionend", () => {
+      this.updateOpacity();
+    });
+
+    window.addEventListener("resize", () => {
+      this.updateOpacity();
+    });
+
+    elements.button[0].addEventListener("click", () => {
+      let segmentWidth = elements.segments[0].offsetWidth;
+      let windowWidth =
+        window.innerWidth || document.documentElement.clientWidth;
+      let lastItemShown = this.checkIfInView(
+        document.querySelector(".timeline .segment:last-of-type")
+      );
+      let offset;
+
+      if (lastItemShown) {
+        // move back to first item
+        offset = 0;
+        this.buttonClickCounter = 0;
+      } else {
+        this.buttonClickCounter++;
+        offset = segmentWidth * this.buttonClickCounter;
+
+        if (windowWidth + offset >= this.timeline.offsetWidth) {
+          // show the last item
+          offset = this.timeline.offsetWidth - windowWidth;
```

Comment:
this probably also needs to take the button's width into account, so there's no overlap:

![image](https://user-images.githubusercontent.com/177243/92638002-7d519d80-f28e-11ea-9eae-78a1d132b31c.png)

Reply:
nice catch!